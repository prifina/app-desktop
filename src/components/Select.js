import React, { forwardRef } from "react";
import styled from "styled-components";
import { space, typography } from "styled-system";
import { default as styledProps } from "@styled-system/prop-types";
import { menuDownIcon as ChevronDown, BlendIcon } from "@blend-ui/icons";
import i18n from "../lib/i18n";
i18n.init();

//import PropTypes from "prop-types";

const selectVariation = props => {
  //console.log("VARIATION ", props);
  let selectProps = props.theme.componentStyles.select;

  return selectProps;
};

const SelectElement = styled.select`
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  margin:0;
  ${selectVariation}
  ${space} ${typography}
  ::-ms-expand {
    display: none;
  }
  &:disabled {
    opacity: 0.25;
  }
`;


const ClickableIcon = styled(BlendIcon).attrs(props => ({
  color: props.theme.colors.black,
}))`
  pointer-events: none;
  margin-left: -32px;
`;

const Select = forwardRef((props, ref) => {
  // console.log("child",props)
  return (
    <React.Fragment>
      <SelectElement {...props} ref={ref} className={props.className} >
          <option value="1"> {i18n.__("listFirstName")}  </option>
        {/*props.child.length > 0 && props.child.map(row=>)
          <option value={row.value}> {i18n.__(row.label)}  </option>
        */}
      </SelectElement>
      
    </React.Fragment>
  );
});

Select.propTypes = {
  ...styledProps.space,
  ...styledProps.typography,
};
Select.displayName = "Select";
Select.isField = true;

export default Select;
