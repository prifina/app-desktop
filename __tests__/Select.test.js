import React, {createRef} from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "@blend-ui/core";
import Select from "../src/components/Select";

test("Link renders correctly", () => {
    const tree = renderer.create(
        <ThemeProvider>
            <Select id="cabinClass" name="cabinClass">
                <option value="1">First Name</option>
            </Select>
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

