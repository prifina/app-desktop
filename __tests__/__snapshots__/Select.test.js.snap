// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1w52cfi-0 fESRGN"
>
  <select
    className="sc-bdfBwQ beJZeV"
    id="cabinClass"
    name="cabinClass"
  >
    <option
      value="1"
    >
       
      First Name
        
    </option>
  </select>
</div>
`;
